static/js/publisher/market/market.js

```diff
@@ -88,32 +105,47 @@ function initForm(config, initialState, errors) {
   initSnapScreenshotsEdit(
     config.screenshotsToolbar,
     config.screenshotsWrapper,
-    state
+    state,
+    (nextState) => {
+      updateState(state, nextState);
+      updateFormState();
+    }
   );
 
-  // when anything is changed update the state
-  marketForm.addEventListener('change', function(event) {
-    let formData = new FormData(marketForm);
+  function checkForm() {
+    const diff = diffState(initialState, state);
+
+    if (diff && isFormValid()) {
+      enableSubmit();
+    } else {
+      disableSubmit();
+    }
+  }
 
+  function updateFormState() {
     // Some extra modifications need to happen for the checkboxes
-    publicMetrics(marketForm, formData);
+    publicMetrics(marketForm);
```

Comment:
This needs to be called before we read FormData, so that blacklisted metrics input is properly populated.